### DISCUSSION TITLE
Download all documents from appwrite collection in React

### DISCUSSION THREAD

**Discussion by thejunghare** [appwrite/appwrite#8268] [Q&A]
In my app-write database I have a collection with documents and I have to download all the documents with a click in my react app I have write the script but the issue is it only download first 26 document. I have around 1000 document and I have to download then all at once

import React from 'react';
import { Databases, Client } from 'appwrite';

const appwrite = new Client();
appwrite
    .setEndpoint('https://cloud.appwrite.io/v1') // Set your Appwrite endpoint
    .setProject('your-project-id'); // Set your project ID

const database = new Databases(appwrite);

async function dumpCollection(collectionId) {
    try {
        // Adjust the listDocuments call to include databaseId and any necessary parameters
        const response = await database.listDocuments('your-database-id', collectionId);
        const documents = response.documents; // Extract documents from the response
        const json = JSON.stringify(documents);
        const blob = new Blob([json], { type: 'application/json' });
        const url = URL.createObjectURL(blob);
        const link = document.createElement('a');
        link.href = url;
        link.download = 'collection.json';
        document.body.appendChild(link);
        link.click();
        document.body.removeChild(link);
      } catch (error) {
        console.error(`Failed to dump collection: ${error.message}`);
      }
    }

    const App = () => {
      return (
        <button onClick={() => dumpCollection('your-collection-id')}>
          Download Collection
        </button>
      );
    }

    export default App;

I have write the script but the issue is it only download first 26 document. I have around 1000 document and I have to download then all at once

**Comment by ItzNotABug** [comment]
You can do one of these -
1. Use `Query.limit()` [not recommended],
2. You pagination using a `while` loop with `Query.cursorAfter` for a better performance.
